### PR TITLE
feat: remove non latest kernel files

### PR DESCRIPTION
**Pull request by obi-t4**
# About this PR

This PR supports to eliminate non latest kernels from the /boot directory.

# Ticket

https://tier4.atlassian.net/browse/T4PB-25530

### CODE REVIEW

File: metadata/ota_metadata/metadata_gen.py [hunk 1]
```diff
@@ -111,6 +117,54 @@ def ignore_rules(target_dir, ignore_file):
...
+    kfiles = sorted(kfile_glob, key=cmp_to_key(compare), reverse=True)
+
+    return Path(kfiles[0])  # latest
+
+
+def _list_non_latest_kernels(boot_dir: Path):
+    kfiles_path = str(boot_dir / "vmlinuz*")
+    ifiles_path = str(boot_dir / "initrd.img*")
+    sfiles_path = str(boot_dir / "System.map*")
+    cfiles_path = str(boot_dir / "config*")
```
Comment (Bodong-Yang) [inline metadata/ota_metadata/metadata_gen.py:142]: I am a little bit worried about using `config*` as pattern to search file under boot_dir, what about using more precise pattern like `config-*.*.*-*-*`.

**Comment by Bodong-Yang** [comment]
About the PR title, I think the actual things we do is to not include the unused kernels and related files(system.map, initrd.img, config) into the generated image meta, is that correct?

**Comment by obi-t4** [comment]
@Bodong-Yang 

> About the PR title, I think the actual things we do is to not include the unused kernels and related files(system.map, initrd.img, config) into the generated image meta, is that correct?

Yes! the latest kernel should be used and updated by the OTA client.

**Review by Bodong-Yang — APPROVED** [review]
LGTM!

**Pull request merged**